app/services/authorizer.rb

```diff
@@ -12,8 +12,9 @@ def can?(permission, subject = nil)
       user.permissions.where(:name => permission).present?
     else
       return true if user.admin?
-      collection = @cache[subject.class.to_s][permission] ||= find_collection(subject.class, :permission => permission)
-      collection.include?(subject)
+      @cache["#{subject.class.to_s}-#{permission}-#{subject.id}"] ||=
```

Comment:
It depends on the action, for most actions a new Authorizer is created so the cache is thrown away (https://github.com/theforeman/foreman/blob/develop/app/helpers/application_helper.rb#L104)